dlt/destinations/dataset/relation.py

```diff
@@ -394,5 +425,24 @@ def __getitem__(self, columns: Sequence[str]) -> Self:
             f" {type(columns).__name__}`. Valid types are: ['Sequence[str]']"
         )
 
-    def head(self, limit: int = 5) -> Self:
-        return self.limit(limit)
+    #
+    # Builtins
+    #
+    def __str__(self) -> str:
+        # TODO: merge detection of "simple" transformation that preserve table schema
+        query_expr = self.__dict__.get("_sqlglot_expression", None)
+        msg = (
+            "Relation"
+            f" query:\n{indent(query_expr.sql(dialect=self.query_dialect(), pretty=True), prefix='  ')}\n"
+        )
+        msg += "Columns:\n"
+        for column in self.columns_schema.values():
+            # TODO: show x-annotation hints
+            msg += f"{indent(column['name'], prefix='  ')} {column['data_type']}\n"
+        return msg
+
+    def __copy__(self) -> Self:
```

Comment:
OK this is more or less what __repr__ should show:
- dataset
- query as text (SQL)